// compitojs/index.js

var users = [
  {
    id : 1 ,
    name : 'Ciccio' ,
    surname : 'Barresi',
    friends : [2],
    posts : [{id : 1 , text : 'ciao'}],
    requestReceived : [] ,
    requestSent : []

  },
  {
    id : 2 ,
    name : 'Luca' ,
    surname : 'Cortile',
    friends : [1],
    posts : [{id : 1 , text : 'nuovo post'}],
    requestReceived : [] ,
    requestSent : []

  },
  {
    id : 3 ,
    name : 'Marco' ,
    surname : 'Velardita',
    posts : [],
    friends : [],
    requestReceived : [4] ,
    requestSent : []

  },
  {
    id : 4 ,
    name : 'Pippo' ,
    friends : [],
    posts : [],
    requestReceived : [] ,
    requestSent : [3]

  },

  {
    id : 5 ,
    name : 'Arturo' ,
    surname : 'Baudo',
    friends : [],
    posts : [],
    requestReceived : [] ,
    requestSent : []

  }
]

exports.getAllUsers = function(){
  return users ;
}

exports.getAllId = function(){
  var allId = []
  for (var i=0; i<users.length; i++) {
    allId.push(users[i].id)
  }
  return allId ;
}

exports.insertUser = function(user){
  user.id = users[users.length-1].id +1;
  return users.push(user)
}

exports.insertPost = function(id , newPost){
  for (var i=0; i<users.length; i++) {
    if(users[i].id === id){
        return users[i].posts.push(newPost)
      }
    }
  }


exports.sendRequest = function(id , id2){
  for (var i=0; i<users.length; i++) {
    if (id2 === users[i].id) {
    users[i].requestReceived.push(id) ;
    }
    if (id === users[i].id) {
    users[i].requestSent.push(id2)
    }
  }
  return 'The request of id '+id+' to id '+id2+' has been sent';
}

exports.acceptRequest = function(id , id2){
  for (var i=0; i<users.length; i++) {
    if(users[i].id === id){
    users[i].friends.push(id2)
    }
    if(users[i].id === id2){
    users[i].friends.push(id)
    }
  }
return 'The request of id '+id+' to id '+id2+' has been accepted';
}








/*
console.log(sendRequest(4,5));
console.log(acceptRequest(4,3));
console.log("Tutti gli id esistenti:",getAllId());
console.log(insertUser({name : 'Giovanni', surname:'Rana',  friends : [],requestSent : [],requestReceived : []}));
console.log(insertPost(3,{text:'ciao'}));
console.log(getAllUsers());
*/
